test/extensions/filters/http/cache/cache_filter_integration_test.cc

```diff
@@ -275,7 +302,55 @@ TEST_P(CacheIntegrationTest, GetRequestWithBodyAndTrailers) {
   }
 }
 
+// Send the same GET request with body and trailers twice, then check that the response
+// doesn't have an age header, to confirm that it wasn't served from cache.
+TEST_P(CacheIntegrationTest, GetRequestWithResponseTrailers) {
+  initializeFilterWithTrailersEnabled(default_config);
+  // Include test name and params in URL to make each test's requests unique.
+  const Http::TestRequestHeaderMapImpl request_headers =
+      httpRequestHeader("GET", /*authority=*/"GetRequestWithResponseTrailers");
+
+  const std::string response_body(42, 'a');
+  Http::TestResponseHeaderMapImpl response_headers = {{":status", "200"},
+                                                      {"date", formatter_.now(simTime())},
+                                                      {"cache-control", "public,max-age=3600"}};
+  const Http::TestResponseTrailerMapImpl response_trailers{{"response1", "trailer1"},
+                                                           {"response2", "trailer2"}};
+  // Send GET request, receive a response from upstream, cache it
+  {
+    IntegrationStreamDecoderPtr response_decoder = sendHeaderOnlyRequestAwaitResponse(
+        request_headers, simulateUpstreamResponse(response_headers, makeOptRef(response_body),
+                                                  makeOptRef(response_trailers)));
+    EXPECT_THAT(response_decoder->headers(), IsSupersetOfHeaders(response_headers));
+    EXPECT_EQ(response_decoder->headers().get(Http::CustomHeaders::get().Age).size(), 0);
```

Comment:
nit: compare the age to "" and then if it fails the test log will tell you what the age actually was, rather just the size of the string.

Reply:
Please, see this discussion on why we decided to check the size here: https://github.com/envoyproxy/envoy/pull/15910#discussion_r612404635 and let me know if you want me to update it further.

Reply:
Oh sorry ... yeah what you have is correct. Optional suggestion to make it slightly more obvious what's going on:
```
  Http::GetResult age_header_values = response_decoder->headers().get(Http::CustomHeaders::get().Age);
  EXPECT_EQ(0, age_header_values.size());
```
Up to you whether you think the readability justifies the extra line.